File: spring-security/session-fixation-attack-protection/src/main/java/com/example/demo/MyController.java
package com.example.demo;

import javax.servlet.http.HttpServletRequest;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@RequestMapping("/")
public class MyController {


    @GetMapping("/")
    public String index(final Model model, final HttpServletRequest req) {
        final String number = (String) req.getSession().getAttribute("number");
        final MyData data = new MyData();
        data.setNumber(number);
        model.addAttribute("data", data);
        return "index";
    }

    @PostMapping("/number")
    public String number(@ModelAttribute final MyData data, final HttpServletRequest req) {
        req.getSession().setAttribute("number", data.getNumber());
        return "redirect:/";
    }

    @GetMapping("/secure")
    public String secure(final Model model, final HttpServletRequest req) {
        final String number = (String) req.getSession().getAttribute("number");
        model.addAttribute("number", number);
        return "secure";
    }
}

File: spring/datetime-to-json/src/main/kotlin/com/example/datetime_to_json/MyController.kt
package com.example.datetime_to_json

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody

@Controller
class MyController {

    @GetMapping
    fun index(model: Model): String {
        model.addAttribute("dto", MyDto.generate())
        return "index"
    }

    @GetMapping("/api")
    @ResponseBody
    fun api(): MyDto {
        return MyDto.generate()
    }
}
